framework/include/utils/MathUtils.h

```diff
@@ -338,6 +338,18 @@ mooseSetToZero(std::vector<Real> & vec)
  */
 std::vector<std::vector<unsigned int>> multiIndex(unsigned int dim, unsigned int order);
 
+template <typename T, typename T2>
+T
+linearInterpolation(
+    T x, const T2 x1, const T2 x2, const T2 y1, const T2 y2, bool derivative = false)
```

Comment:
I'm not a fan of these `bool` parameters to get a derivative. Why not make a new function here?

Reply:
For example, if there is a subclass in `linearInterpolation` (and the math works out), I can pass through the `derivative` bool and don't have to do anything special.

Reply:
> Those bool parameters are considered bad style. If you want to keep one function I'd recommend the last parameter to be a required enum class.

Looks like we were thinking about it at the same time...

Reply:
Enum is overkill since it's either a derivative or it's not, right? I suppose you could do second derivatives, but really only the first derivative is needed for jacobians or newton-rhapson iterations.

Reply:
No, it is not "overkill", it makes the function call more readable. Otherwise a passeby will have no idea what the difference between.

```
MathUtils::linearInterpolation(0.15, 0.1, 0.2, 0.4, 1.2);
```
 and

```
MathUtils::linearInterpolation(0.15, 0.1, 0.2, 0.4, 1.2, true);
```

is.

Reply:
Okay...it's the better way, you're right...